src/main/scala/dspjunctions/JtagAxi.scala

```diff
@@ -0,0 +1,126 @@
+// See LICENSE for license details.
+
+package dspjunctions
+
+import cde._
+import chisel3._
+import chisel3.util._
+import jtag._
+import junctions._
+
+case object DspChainIncludeJtag extends Field[Bool]
+
+class DspJtagIO extends JtagIO {
+  // val TRST = Input(Bool())
+}
+
+trait HasDspJtagParameters {
+  implicit val p: Parameters
+  val includeJtag = try {
+    if (p(DspChainIncludeJtag)) {
+      true
+    } else {
+      false
+    }
+  } catch {
+    case _: ParameterUndefinedException => false
+  }
+}
+
+trait HasDspJtagIO extends HasDspJtagParameters {
+  val p: Parameters
+  val jtag = if (includeJtag) {
+      Some(new DspJtagIO)
+  } else {
+    None
+  }
+}
+
+trait HasDspJtagModule extends HasDspJtagParameters {
+  val io: HasDspJtagIO
+
+  val jtagMaster = if (includeJtag) {
+    Some(Module(new JtagAxiMaster()))
+  } else {
+    None
+  }
+
+  def jtagCtrlAxiMasters: Seq[NastiIO] = jtagMaster.map(m => Seq(m.io.ctrlAxi)).getOrElse(Seq())
+  def jtagDataAxiMasters: Seq[NastiIO] = jtagMaster.map(m => Seq(m.io.ctrlAxi)).getOrElse(Seq())
+
+  def jtagConnect: Unit = (io.jtag, jtagMaster) match {
+    case (Some(jtag), Some(master)) =>
+      jtag <> master.io.jtag
+    case (None, None) =>
+    case _ => throw dsptools.DspException("One of jtagMaster and io.jtag existed, but the other did not!")
+  }
+}
+
+class JtagAxiMaster()(implicit p: Parameters) extends NastiModule()(p) {
+  val io = IO(new Bundle {
+    val jtag    = new DspJtagIO
+    val ctrlAxi = new NastiIO
+    val dataAxi = new NastiIO
+  })
+
+  val tap = Module(new JtagTwoAxiMasterClocked(io.jtag.TCK.asClock, false.B)) // io.jtag.TRST))
+  io.jtag <> tap.io.jtag
+  tap.io.control.fsmAsyncReset := false.B
+
+  io.ctrlAxi <> tap.io.ctrlAxi
+  io.dataAxi <> tap.io.dataAxi
+}
+
+class JtagTwoAxiMasterClocked(modClock: Clock, modReset: Bool)(implicit val p: Parameters)
+  extends Module(override_clock=Some(modClock), override_reset=Some(modReset))
```

Comment:
I want to deprecate this style, consider using the withClockAndReset style in the builtin-debuggers example, which has a withClockAndReset block inside the system clock/reset domain. It also had the clock crossings in the JTAG block scope, which I think makes logical sense (in terms of encapsulating JTAG stuff).

Though in the long term we'll probably want FIRRTL to hook up JTAG connections in the background, since this is just plumbing busywork.